### DISCUSSION TITLE
Can pypowerwall be used to set state on your Powerwalls like cycle between time of use and backup only mode or is it just for monitoring?

### DISCUSSION THREAD

**Discussion by mikkyo9** [jasonacox/Powerwall-Dashboard#15] [Q&A]
It seems like there is a strong demand to be able to switch between self powered, using your Solar/PW during only the off peak times, and only ever switching to Time Of Use during the peak pricing, so you can really export when its most cost-worthy.
Sort of like the old Balanced Mode the App used to offer.
Can pypowerwall send api requests or is it only for monitoring(get)?
There is a powewallCloud setup that lets you do this(requires hacking around getting token), but seems a bit excessive when it could maybe just be done through your tool.

**Comment by jasonacox** [comment]
Great question, @mikkyo9 !  pyPowerwall uses the customer credentials to get access to the APIs to *GET* data.  To be able to *POST* changes, basically do what you are wanting, requires additional access level (a token).  I started working on how we can do that to mimic what the smartphone app is able to do.  Once we have that, we can use pyPowerwall to send commands to the system as well as scrape data.  To your point, the methods to get the token are non-trivial and apparently have changed over the years, indicating that they could change again. If anyone has any ideas, please feel free to post here and I'll link to the pyPowerwall project.
